apps/backend/src/datasources/postgres/TechniqueDataSource.ts

```diff
@@ -224,6 +225,90 @@ export default class PostgresTechniqueDataSource
     }
   }
 
+  async assignProposalToTechniques(
+    proposalPk: number,
+    techniqueIds: number[]
+  ): Promise<boolean> {
+    const checks = await Promise.all(
```

Comment:
My suggestions are:
- If I’m correct, we need to remove the rows that are no longer selected
- You can use a single SQL query to check if a row exists before inserting:
```knex.raw('NSERT INTO … SELECT … WHERE NOT EXISTS ( SELECT 1 …```
- please use a transaction
If you need any help, please let me know